### PR TITLE
O11Y-3736: opentelemetry-dart: Deprecate MeterSharedState

### PR DESCRIPTION
**Pull request by michaelyeager-wf**
## Which problem is this PR solving?

MeterSharedState currently encapsulates actions which should be the responsibility of MeterProvider.

## Short description of the change

In order to improve adherence to the OpenTelemetry specification, this PR moves state management responsibility for meters into `MeterProvider` and removes shared state classes.

## How Has This Been Tested?

* Confirmed that unit tests pass following these changes.

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Comment by michaelyeager-wf** [comment]
Since we haven't implemented MetricExporters, etc yet we won't be able to use a test application to verify these changes.  The existing unit tests should be sufficient, since they verify that MeterProviders can be created and properly manage state.   QA +1

File: lib/src/sdk/metrics/meter.dart [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
+import '../../experimental_api.dart' as api;
 
 class Meter implements api.Meter {
-  // ignore: unused_field
-  final MeterSharedState _state;
+  final sdk.Resource _resource; // ignore: unused_field
+  final sdk.InstrumentationScope _instrumentationScope; // ignore: unused_field
 
-  Meter(this._state);
+  Meter(this._resource, this._instrumentationScope);
```
Comment (blakeroberts-wk) [inline lib/src/sdk/metrics/meter.dart:12]: I think this should be marked protected

Comment (michaelyeager-wf) [reply lib/src/sdk/metrics/meter.dart:12]: Why?  We don't do that for `Tracer`.

**Review by kennytrytek-wf — APPROVED** [review]
⛄

**Review by blakeroberts-wk — APPROVED** [review]

**Comment by michaelyeager-wf** [comment]
Unit tests pass following this change.

QA +1

**Comment by michaelyeager-wf** [comment]
@Workiva/release-management-p

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**